### PR TITLE
mostly wrks

### PR DESCRIPTION
**Pull request by IRog**
Screen Shot 2020-11-08 at 4 18 54 PM

Sometimes I was getting weird/not goat looking things but I'm not sure if it's the `config.toml` optimization things, something with bevy, or, other? `CXX=$(which clang++)` was to fix https://github.com/gfx-rs/gfx/issues/2520#issuecomment-497955099

### CODE REVIEW

File: src/ccall/.cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[target.x86_64-unknown-linux-gnu]
+linker = "/usr/bin/clang"
+rustflags = ["-Clink-arg=-fuse-ld=lld", "-Zshare-generics=y"]
+
+# NOTE: you must manually install https://github.com/michaeleisel/zld on mac. you can easily do this with the "brew" package manager:
```
Comment (pdlla) [inline src/ccall/.cargo/config.toml:5]: cool :O

File: src/ccall/src/main.rs [hunk 2]
```diff
@@ -1,88 +1,90 @@
...
+    );
+    mesh.set_indices(Some(Indices::U32(f.to_vec())));
+
+    let mesh_handle = meshes.add(mesh);
+    let material_handle = materials.add(StandardMaterial {
+        albedo: Color::rgb(0.8, 0.7, 0.6),
+        ..Default::default()
+    });
+
+    commands
```
Comment (pdlla) [inline src/ccall/src/main.rs:72]: :O cool

**Review by pdlla — APPROVED** [review]
🐐🐐🐐

File: src/ccall/src/goat.rs [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-use std::fs::{File};
-use std::path::Path;
-use std::io::LineWriter;
+use std::fs::File;
 use std::io::prelude::*;
+use std::io::LineWriter;
+use std::path::Path;
 
 //#[link(name = "animalclub")]
+#[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (IRog) [reply src/ccall/src/goat.rs:11]: I was just removing compiler warnings from unused functions that bothered me before

File: src/ccall/src/goat.rs [hunk 4]
```diff
@@ -105,10 +102,21 @@ pub fn write_obj_from_buffers(p: &[f32], n: &[f32], tc: &[f32], f: &[u32]) -> st
...
+                x[0] + 1,
+                x[1] + 1,
+                x[1] + 1,
+                x[1] + 1,
+                x[2] + 1,
+                x[2] + 1,
+                x[2] + 1
+            )
+            .as_bytes(),
+        )?;
```
[the remark this answers is not in the document]
Comment (IRog) [reply src/ccall/src/goat.rs:119]: Rustfmt

**Pull request merged**